testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,711 @@
+                writer.println(ff);
+            }
+            writer.close();
+
+            writer = new PrintWriter(rootFold + "/passing_order.txt", "UTF-8");
+            for (String key : nameToInstance.keySet()) {
+                writer.println(key);
+            }
+            writer.close();
+        } else if (state.equals("failing_order")) {
+            writer = new PrintWriter(Configuration.config().getProperty("replay.allFieldsFold") + "/failing_order.txt", "UTF-8");
+            for (String ff : allFieldName) {
+                writer.println(ff);
+            }
+            writer.close();
+
+            writer = new PrintWriter(rootFold + "/failing_order.txt", "UTF-8");
+            for (String key : nameToInstance.keySet()) {
+                writer.println(key);
+            }
+            writer.close();
+        }
+    }
+
+    /**
+     * Adds the current loadable classes to the current class list in eagerLoadingFields.txt file in the phase 2tmp.
+     * @throws IOException
+     */
+    public void capture_class() throws IOException {
+        // read whitelist;
+        try (BufferedReader br = new BufferedReader(new FileReader(Configuration.config().getProperty("replay.pkgFile")))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                whiteList.add(line);
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            return;
+        }
+
+        // get all loadable classes
+        Class[] loadedClasses = MainAgent.getInstrumentation().getAllLoadedClasses();
+        Set<String> classes = new HashSet<>();
+        for (Class c : loadedClasses) {
+            String clz = c.getName();
+            if ((clz.contains("java.") && !clz.startsWith("java.lang.System"))
+                    || (clz.contains("javax.") && !clz.startsWith("javax.cache.Caching"))
+                    || clz.contains("javafx.")
+                    || clz.contains("jdk.")
+                    || clz.contains("scala.")
+                    || clz.contains("sun.")
+                    || clz.contains("edu.illinois.cs")
+                    || clz.contains("org.custommonkey.xmlunit")
+                    || clz.contains("org.junit")
+                    || clz.contains("statecapture.com.")
+                    || clz.contains("statecapture.org.")) {
+                continue;
+            }
+            classes.add(clz);
+        }
+
+        // write to the eagerLoadingFields.txt file
+        PrintWriter writer = new PrintWriter(rootFold + "/" + "eagerLoadingFields.txt", "UTF-8");
+        for (String str : classes) {
+            writer.println(str);
+        }
+        writer.close();
+
+    }
+
+    protected boolean shouldCapture(Field f) {
+        return true;
+    }
+
+    String createSubxmlFold() {
+        String state = Configuration.config().getProperty("statecapture.state");
+        String subxmlDir = "";
+        if (state.equals("passing_order")) {
```

Comment:
Do we really need to explicitly distinguish between passing_order and failing_order? The logic here doesn't need to know whether order is passing or failing in order to capture state and serialize it. This distinction seems only relevant for determining name of directory to write to, but that can simply be passed in as an explicit input.